sw/device/lib/runtime/ibex.h

```diff
@@ -26,19 +26,17 @@
  * V20191213, pp. 61.
  */
 inline uint64_t ibex_mcycle_read(void) {
-  uint32_t cycle_low = 0;
-  uint32_t cycle_high = 0;
-  uint32_t cycle_high_2 = 0;
-  asm volatile(
-      "read%=:"
-      "  csrr %0, mcycleh;"     // Read `mcycleh`.
-      "  csrr %1, mcycle;"      // Read `mcycle`.
-      "  csrr %2, mcycleh;"     // Read `mcycleh` again.
-      "  bne  %0, %2, read%=;"  // Try again if `mcycle` overflowed before
-                                // reading `mcycleh`.
-      : "+r"(cycle_high), "=r"(cycle_low), "+r"(cycle_high_2)
-      :);
-  return (uint64_t)cycle_high << 32 | cycle_low;
+  for (;;) {
```

Comment:
I'm still not sure about replacing the inline asm block here. At O0 this might generate rather more code than expected. You could reasonably argue whether you worry about mcycle_read being introducing its own noise (due to lengthier execution time) at that point though...

Reply:
I think I agree with Alex here. This code will never be run on the host.

Reply:
Reverted. I think it would be OK even with `-O0`, but as you say there isn't much benefit to changing it.